salt/thorium/__init__.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+'''
+The thorium system allows for advanced event tracking and reactions
+'''
+# Needed:
+# Use a top file to load sls files locally
+# use the existing state system to compile a low state
+# Create a new state runtime to run the low state flow programming style
+# Create the thorium plugin system
+# Add dynamic recompile of thorium ruleset on given interval
+
+# Import python libs
+from __future__ import absolute_import
+import time
+import logging
+import traceback
+
+# Import Salt libs
+import salt.state
+from salt.exceptions import SaltRenderError
+
+log = logging.getLogger(__name__)
+
+
+class ThorState(salt.state.HighState):
+    '''
+    Compile the thorium state and manage it in the thorium runtime
+    '''
+    def __init__(self, opts):
+        opts['file_roots'] = opts['thorium_roots']
+        self.opts = opts
+        salt.state.HighState.__init__(self, self.opts, loader='thorium')
+        self.state.inject_globals = {'__reg__': {}}
+        self.event = salt.utils.event.get_master_event(
+                self.opts,
+                self.opts['sock_dir'])
+
+    def start_runtime(self):
+        '''
+        Start the system!
+        '''
+        chunks = self.get_chunks()
+        while True:
+            try:
+                self.call_runtime(chunks)
+            except Exception:
+                time.sleep(self.opts['thorium_interval'])
+
+    def get_chunks(self, exclude=None, whitelist=None):
+        '''
+        Compile the top file and return the lowstate for the thorium runtime
+        to iterate over
+        '''
+        ret = {}
+        err = []
+        try:
+            top = self.get_top()
+        except SaltRenderError as err:
+            return ret
+        except Exception:
+            trb = traceback.format_exc()
```

Comment:
We should also log the exception message and show the traceback at the debug level(`log.error(msg, exc_info_on_loglevel=logging.DEBUG)`)